enterprise/views.py

```diff
@@ -714,6 +750,13 @@ def post(self, request, enterprise_uuid, course_id):
                     enterprise_customer_user=enterprise_customer_user,
                     course_id=course_id,
                 )
+
+            try:
+                # Post to E-Commerce for audit enrollment.
+                self.post_order_to_ecommerce(request, selected_course_mode['sku'])
+            except (HTTPError, Timeout, HttpClientError):
+                return redirect('enterprise_course_enrollment_page', enterprise_uuid, course_id)
+
             client = EnrollmentApiClient()
```

Comment:
I'm wondering if we'll run into the same race condition that caused ENT-512 by keeping this code around. My understanding was that the ecommerce call was queuing a task that would subsequently do some work, including calling back to the enrollment api in lms. Because we were also directly calling the enrollment api in the same view, it was causing an error triggered by the potential race condition. @douglashall what do you think?